packages/contracts/contracts/L1/ValidatorPool.sol

```diff
@@ -316,23 +424,18 @@ contract ValidatorPool is ReentrancyGuardUpgradeable, Semver {
      * @return The address of the validator.
      */
     function nextValidator() public view returns (address) {
-        uint256 nextOutputIndex = L2_ORACLE.nextOutputIndex();
-        if (nextOutputIndex > 0) {
-            // TODO(pangssu): Make sure to use the finalized output root.
-            Types.CheckpointOutput memory output = L2_ORACLE.getL2Output(nextOutputIndex - 1);
-            uint256 rand = uint256(
-                keccak256(
-                    abi.encodePacked(
-                        validators.length,
-                        output.outputRoot,
-                        output.submitter,
-                        output.timestamp,
-                        output.l2BlockNumber
-                    )
-                )
-            ) % validators.length;
-
-            return validators[rand];
+        if (nextPriorityValidator != address(0)) {
+            uint256 l2BlockNumber = L2_ORACLE.nextBlockNumber();
+            uint256 l2Timestamp = L2_ORACLE.computeL2Timestamp(l2BlockNumber + 1);
```

Comment:
If the time for the actual L2 block was increased by 1, wouldn't it make sense to essentially modify the computeL2Timestamp method?